port/cpl_aws.cpp

```diff
@@ -2298,6 +2494,53 @@ void VSIS3HandleHelper::RefreshCredentials(const std::string &osPathForOption,
             m_osSessionToken = std::move(osSessionToken);
         }
     }
+    else if (m_eCredentialsSource == AWSCredentialsSource::CREDENTIAL_PROCESS)
+    {
+        // First, check if we have a cached credential_process command
+        std::string osCredentialProcess;
+        {
+            CPLMutexHolder oHolder(&ghMutex);
+            osCredentialProcess = gosCredentialProcessCommand;
+        }
+
+        // If no cached command, re-read config files to get the credential_process setting
+        if (osCredentialProcess.empty())
+        {
+            std::string osSecretAccessKey, osAccessKeyId, osSessionToken,
+                osRegion;
+            std::string osCredentials, osRoleArn, osSourceProfile, osExternalId;
+            std::string osMFASerial, osRoleSessionName, osWebIdentityTokenFile;
+            std::string osSSOStartURL, osSSOAccountID, osSSORoleName,
+                osSSOSession;
+
+            if (GetConfigurationFromAWSConfigFiles(
+                    osPathForOption, nullptr, osSecretAccessKey, osAccessKeyId,
+                    osSessionToken, osRegion, osCredentials, osRoleArn,
+                    osSourceProfile, osExternalId, osMFASerial,
+                    osRoleSessionName, osWebIdentityTokenFile, osSSOStartURL,
+                    osSSOAccountID, osSSORoleName, osSSOSession,
+                    osCredentialProcess))
+            {
+                // Cache the command for future use
+                CPLMutexHolder oHolder(&ghMutex);
+                gosCredentialProcessCommand = osCredentialProcess;
+            }
+        }
+
+        // Now use the cached refresh mechanism
+        if (!osCredentialProcess.empty())
```

Comment:
I believe all of his should be deleted, so as to be "symetric" with the other credential soures